bee-message/src/payload/transaction/essence/regular.rs

```diff
@@ -234,12 +235,25 @@ impl RegularEssenceBuilder {
             }
         }
 
-        // TODO
-        // inputs.sort();
-        // outputs.sort();
+        // Inputs must be lexicographically sorted in their serialised form
+        let packed_inputs: Result<Vec<_>, _> = self.inputs.iter().map(|input| {
+            let mut packed_input = vec![];
+            input.pack(&mut packed_input).and(Ok(packed_input))
```

Comment:
There is `pack_new` that returns a new filled vector :)

Reply:
Didn't see that one there, makes things much cleaner!